src/ucp/core/ucp_ep.c

```diff
@@ -3253,14 +3254,17 @@ ucp_ep_is_am_keepalive(ucp_ep_h ep, ucp_rsc_index_t rsc_idx)
     }
 
     wiface = ucp_worker_iface(ep->worker, rsc_idx);
-    return ucs_test_all_flags(wiface->attr.cap.flags,
+    return (lane != UCP_NULL_LANE) &&
+           !(ucp_ep_config(ep)->p2p_lanes & UCS_BIT(lane)) &&
```

Comment:
1. can we add ucp_ep_config(ep)->is_am_keepalive and set it during ucp_ep_config_init?
2. do we still need to check UCT_IFACE_FLAG_CONNECT_TO_IFACE below?

Reply:
> can we add ucp_ep_config(ep)->is_am_keepalive and set it during ucp_ep_config_init?

we couldn't set `is_am_keepalive` during `ucp_ep_config_init`, since we should use ep_check in case of REMOTE_ID hasn't been resolved yet.

> do we still need to check UCT_IFACE_FLAG_CONNECT_TO_IFACE below?

yes, we could avoid checking it